ouroboros-network/src/Ouroboros/Network/Protocol/BlockFetch/Client.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Ouroboros.Network.Protocol.BlockFetch.Client where
+
+import Protocol.Core
+import Pipes (Producer)
+import qualified Pipes
+
+import Ouroboros.Network.Protocol.BlockFetch.Type
+
+{-------------------------------------------------------------------------------
+  Client stream of @'BlockFetchClientProtocol'@ protocol
+-------------------------------------------------------------------------------}
+
+data BlockFetchClientSender range m a where
+
+  -- | Send a request for a range of blocks.
+  --
+  SendBlockRequestMsg
```

Comment:
Maybe it's nit-picking, but this isn't a message, and nothing is being sent, so it shouldn't include "send" or "msg".

Reply:
That's ok, good names aren't easy to get right, and I am happy to take advise on that.  In `ChainSync.Client.ClientStIdle` names start with `SendMsg`, maybe that's better than `Send...Msg` since it conveys the purpose of this constructor (it will send a message when the corresponding `Peer` is run).  But I can also remove `Send` and `Msg`; What about `BlockRangeRequest`?